src/test/clojure/clara/test_rules.clj

```diff
@@ -5025,3 +5029,33 @@
     (is (= (query end-session-equal-facts lousy-weather-query)
            (query end-session-identical-facts lousy-weather-query)
            [{}]))))
+
+(deftest test-alpha-batching-with-multiple-ancestors
+  ;; Validate that two distinct fact types are batched together when passed to a condition on a
+  ;; common ancestor type. We use an accumulator because it provides a convenient way to directly
+  ;; access the facts passed to the condition without exposing Clara internals.
+  ;; see issue 257 for more details of the bug surrounding this test
+  (let [accum-state (atom #{})
```

Comment:
With the new approach of adding all arguments provided to the convert-return-fn to this I think it would be more straightforward/obviously correct to use a sequence rather than a set to avoid discarding any entries.  Admittedly in this case it is safe to assume that a 1-item entry can only be replaced by another 1-item entry but I'd rather just avoid anyone reading the code having to think about it.